### PR TITLE
[JENKINS-72796] stable context classloader for Computer.threadPoolForRemoting

### PR DESCRIPTION
**Pull request by jtnord**
Whilst the threadpool used reset the context classloader at the end of any task, it did not ensure that the initial classloader used was anything sepcific, rather it would use whatever the calling threads contextClassLoader was.

This is now fixed as we use the Jenkins WebApp classloader (same as the Timer) which is used by (A)PeriodicTasks.

Whilst we should really not have a context classloader (aka null) and this should be set where needed by code, almost everywhere in Jenkins the context classloader is already the webapp classloader, and so setting this to be different depending on how things where called would seemingly be a little scary.  Arguably this and other context classloaders should be all set to null and any code that wants different should be changed, but this is a larger piece of work that would have potential impact on an unknown number of plugins in the ecosystem, so this fix uses what was set > 90% of the time.

See [JENKINS-72796](https://issues.jenkins.io/browse/JENKINS-72796).

### Proposed changelog entries

- Ensure threads in the `Computer.threadPoolForRemoting`executor service always have the Jenkins webapp `ClassLoader` set as the context `ClassLoader` to prevent random class loading issues when code is running in this `ExecutorService`.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by jglick — APPROVED** [review]

File: core/src/test/java/hudson/model/ComputerTest.java
Comment (timja) [inline core/src/test/java/hudson/model/ComputerTest.java:79]:
```suggestion
    private static class WaitAndGetContextClassLoader {
```

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by rsandell — APPROVED** [review]

**Comment by kmartens27** [comment]
Hi @jtnord, would there be a better version of the changelog entry for this PR? If there should not be a changelog entry, we can definitely adjust, but if there should be something included, would you be able to clarify a bit what that would be?

**Comment by jtnord** [comment]
> Hi @jtnord, would there be a better version of the changelog entry for this PR? If there should not be a changelog entry, we can definitely adjust, but if there should be something included, would you be able to clarify a bit what that would be?

Hi @kmartens27 thanks for the reminder - I have updated the suggested changelog entry.

> - JENKINS-72796, Ensure the `Computer.threadPoolForRemoting` executor always has the Jenkins webapp `ClassLoader` set as the [context `ClassLoader`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Thread.html#getContextClassLoader()) to prevent random class loading issues when code is running in this `ExecutorService`.

**Comment by kmartens27** [comment]
Thank you so much!

**Pull request merged**
